### ISSUE TITLE
"PartialSp" is visible in pinyin options despite of it has been set to hidden

### ISSUE THREAD

**Issue by rocka** [fcitx/fcitx5-chinese-addons#192]
The option `partialSp` is decalred as `OptionWithAnnotation<bool, OptionalHideInDescription>`

https://github.com/fcitx/fcitx5-chinese-addons/blob/64e2a072e92ca06b08b9fd82d7549f86cc0699a9/im/pinyin/pinyin.h#L117-L119

and set to hidden in `pyConfig_`

https://github.com/fcitx/fcitx5-chinese-addons/blob/64e2a072e92ca06b08b9fd82d7549f86cc0699a9/im/pinyin/pinyin.cpp#L785

but it's still vislble in pinyin options:

![image](https://github.com/user-attachments/assets/7a68ba8c-ba46-441e-87ef-218543435b56)

---

- fcitx5 5.1.10
- fcitx5-chinese-addons 5.1.6
- fcitx5-configtool 5.1.6

**Comment by rocka** [comment]
After some debugging (on Android), I found that `option->skipDescription()` always returns `false` when dumping description of `pyConfig_.fuzzyConfig`

https://github.com/fcitx/fcitx5/blob/9ab829b2587c0dd616f92c85068bcb8c220b628b/src/lib/fcitx-config/configuration.cpp#L47-L49

**Comment by wengxt** [comment]
Did some more investigation, looks like a design flaw in fcitx::Configuration.

Option that wraps a config relies on the subConfigSkeleton() to generate description.

However subConfigSkeleton will always return a copied fcitx::Configuration from "defaultValue_".

The reason for this design is that, subConfigSkeleton not only returns the directly wrapped Configuration subclass, but also strip vector from vector<Configuration>. In the vector case, there is no Configuration object, so it has to be a new instance.

Copy constructor defined with FCITX_CONFIGURATION, will only copy the option value and leave all the annotation untouched.

So in short, two issues:
1. subConfigSkeleton always return a new instance. The function signature is std::unique_ptr, so unfortunately, it has to be a new instance for now. One possible way is solve this is to create a new Configuration and do a forwarding from existing object, instead of return a real new instance.
2. the value returned by subConfigSkeleton only copies values. In the past, we are having a similar issue for defaultValue. At that time, syncDefaultValueToCurrent is introduced. Maybe, we can introduce a new method to copy annotation value too.
